### PR TITLE
Change CreateTablet implemention

### PR DESCRIPTION
**Pull request by chaoyli**

### CODE REVIEW

File: be/src/olap/base_compaction.cpp [hunk 1]
```diff
@@ -309,13 +309,13 @@ OLAPStatus BaseCompaction::_do_base_compaction(VersionHash new_base_version_hash
...
-                                    _tablet->num_short_key_fields(), _tablet->num_rows_per_row_block(),
-                                    _tablet->compress_kind(), _tablet->bloom_filter_fpp()};
-    RowsetBuilder* builder = new AlphaRowsetBuilder(); 
+    RowsetWriterContext context = {_tablet->partition_id(), _tablet->tablet_id(),
+                                   _tablet->schema_hash(), rowset_id, 
+                                   RowsetTypePB::ALPHA_ROWSET, _tablet->rowset_path_prefix(),
+                                   _tablet->tablet_schema(), _tablet->num_key_fields(),
+                                   _tablet->num_short_key_fields(), _tablet->num_rows_per_row_block(),
+                                   _tablet->compress_kind(), _tablet->bloom_filter_fpp()};
+    RowsetWriter* builder = new AlphaRowsetWriter();
```
Comment (yiguolei) [inline be/src/olap/base_compaction.cpp]: Maybe should create a rowsetwriterfactory to create different rowsetwrite according to the rowset type.

File: be/src/olap/delta_writer.cpp [hunk 2]
```diff
@@ -113,13 +113,13 @@ OLAPStatus DeltaWriter::init() {
             .set_rowset_state(PREPARING)
             .set_txn_id(_req.txn_id)
             .set_load_id(_req.load_id);
-    RowsetBuilderContext builder_context = context_builder.build();
+    RowsetWriterContext builder_context = context_builder.build();
```
Comment (yiguolei) [inline be/src/olap/delta_writer.cpp]: if the class name is RowsetWriter then the param name should be writer_context?

File: be/src/olap/merger.cpp [hunk 3]
```diff
@@ -33,7 +33,7 @@ using std::vector;
 
 namespace doris {
 
-Merger::Merger(TabletSharedPtr tablet, RowsetBuilder* builder, ReaderType type) : 
+Merger::Merger(TabletSharedPtr tablet, RowsetWriter* builder, ReaderType type) :
```
Comment (yiguolei) [inline be/src/olap/merger.cpp]: writer?

File: be/src/olap/merger.h [hunk 4]
```diff
@@ -30,7 +30,7 @@ class ColumnData;
 class Merger {
 public:
     // parameter index is created by caller, and it is empty.
-    Merger(TabletSharedPtr tablet, RowsetBuilder* builder, ReaderType type);
+    Merger(TabletSharedPtr tablet, RowsetWriter* builder, ReaderType type);
```
Comment (yiguolei) [inline be/src/olap/merger.h]: writer?

File: be/src/olap/merger.h [hunk 5]
```diff
@@ -46,7 +46,7 @@ class Merger {
     }
 private:
     TabletSharedPtr _tablet;
-    RowsetBuilder* _builder;
+    RowsetWriter* _builder;
```
Comment (yiguolei) [inline be/src/olap/merger.h]: writer?

File: be/src/olap/base_compaction.cpp (hunk 1 above)
Comment (kangpinghuang) [inline be/src/olap/base_compaction.cpp]: use shared_ptr or you will delete the pointer in any return place

File: be/src/olap/base_compaction.cpp [hunk 6]
```diff
@@ -309,13 +309,13 @@ OLAPStatus BaseCompaction::_do_base_compaction(VersionHash new_base_version_hash
...
-                                    _tablet->tablet_schema(), _tablet->num_key_fields(),
-                                    _tablet->num_short_key_fields(), _tablet->num_rows_per_row_block(),
-                                    _tablet->compress_kind(), _tablet->bloom_filter_fpp()};
-    RowsetBuilder* builder = new AlphaRowsetBuilder(); 
+    RowsetWriterContext context = {_tablet->partition_id(), _tablet->tablet_id(),
+                                   _tablet->schema_hash(), rowset_id, 
+                                   RowsetTypePB::ALPHA_ROWSET, _tablet->rowset_path_prefix(),
+                                   _tablet->tablet_schema(), _tablet->num_key_fields(),
+                                   _tablet->num_short_key_fields(), _tablet->num_rows_per_row_block(),
+                                   _tablet->compress_kind(), _tablet->bloom_filter_fpp()};
```
Comment (kangpinghuang) [inline be/src/olap/base_compaction.cpp]: I have written a RowserWriterContextBuilder, you can use it

File: be/src/olap/cumulative_compaction.h [hunk 7]
```diff
@@ -172,7 +172,7 @@ class CumulativeCompaction {
     VersionHash _cumulative_version_hash;
     // 新cumulative文件对应的olap index
     RowsetSharedPtr _rowset;
-    RowsetBuilder* _builder;
+    RowsetWriter* _builder;
```
Comment (kangpinghuang) [inline be/src/olap/cumulative_compaction.h]: use std::shared_ptr

**Review by kangpinghuang — CHANGES_REQUESTED** [review]

File: be/src/olap/delta_writer.cpp (hunk 2 above)
Comment (chaoyli) [reply be/src/olap/delta_writer.cpp]: OK

**Review by yiguolei — APPROVED** [review]

**Pull request merged**